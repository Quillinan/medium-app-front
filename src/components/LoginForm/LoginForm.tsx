import { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '../../auth/useAuth';

const LoginForm: React.FC = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');
  const { setAuth } = useAuth();
  const navigate = useNavigate();
  const auth = sessionStorage.getItem('isAuthenticated');

  const login = (auth: string | null) => {
    console.log(auth);
    if (auth) {
      setError('');
      navigate('/home');
    }
  };

  const handleLogin = (e: React.FormEvent) => {
    e.preventDefault();

    const validEmail = 'teste@test.com';
    const validPassword = '1234';

    if (email === validEmail && password === validPassword) {
      setAuth(null);
      login('true');
    } else {
      setError('UsuÃ¡rio ou senha incorretos');
    }
  };

  useEffect(() => {
    login(auth);
  }, [auth, login]);

  return (
    <form action='#' method='POST' onSubmit={handleLogin} className='space-y-6'>
      <div>
        <label
          htmlFor='email'
          className='block text-sm font-medium leading-6 text-gray-900'
        >
          Email address
        </label>
        <div className='mt-2'>
          <input
            id='email'
            name='email'
            type='email'
            required
            autoComplete='email'
            onChange={e => setEmail(e.target.value)}
            className='block w-full rounded-md border-0 py-2 px-3 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
          />
        </div>
      </div>

      <div>
        <label
          htmlFor='password'
          className='block text-sm font-medium leading-6 text-gray-900'
        >
          Password
        </label>
        <div className='mt-2'>
          <input
            id='password'
            name='password'
            type='password'
            required
            autoComplete='current-password'
            onChange={e => setPassword(e.target.value)}
            className='block w-full rounded-md border-0 py-2 px-3 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
          />
        </div>
      </div>

      {error && <p className='text-red-500 text-sm'>{error}</p>}

      <div className='flex flex-col items-center space-y-4'>
        <button
          type='submit'
          className='flex w-1/2 justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-md hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600'
        >
          Sign in
        </button>
      </div>
    </form>
  );
};

export default LoginForm;
