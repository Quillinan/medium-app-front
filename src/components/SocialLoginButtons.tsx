const SocialLoginButtons: React.FC = () => {
  return (
    <div className='flex flex-col space-y-4'>
      <button
        className='flex w-full justify-center rounded-md bg-blue-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-blue-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-600'
        onClick={() => alert('Continue with Azure')}
      >
        Login with Azure
      </button>
      <button
        className='flex w-full justify-center rounded-md bg-gray-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-gray-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-600'
        onClick={() => alert('Continue with GitHub')}
      >
        Login with GitHub
      </button>
    </div>
  );
};

export default SocialLoginButtons;
