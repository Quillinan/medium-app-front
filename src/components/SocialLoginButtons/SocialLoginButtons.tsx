import { useMsal } from '@azure/msal-react';
import microsoft from '/microsoft.svg';
import { useAuth } from '../../auth/useAuth';

const SocialLoginButtons: React.FC = () => {
  const { instance } = useMsal();
  const { setAuth } = useAuth();

  const handleAzureLogin = () => {
    instance
      .loginPopup({
        scopes: ['user.read'],
      })
      .then(response => {
        const token = response.uniqueId;
        if (token) {
          setAuth(token);
          console.log('Autenticado com sucesso', response);
        }
      })
      .catch(error => {
        console.error('Erro de autenticação:', error);
      });
  };
  return (
    <div className='flex flex-col items-center space-y-4'>
      <button
        className='flex items-center w-1/2 justify-center rounded-md bg-white border border-black/20 px-3 py-1.5 text-sm font-semibold leading-6 text-black shadow-md hover:bg-gray-100 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-600'
        onClick={handleAzureLogin}
      >
        <img src={microsoft} alt='microsoft icon' className='mr-4 h-5 w-5' />
        Microsoft
      </button>

      {/* <button
        className='flex w-full justify-center rounded-md bg-gray-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-gray-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-gray-600'
        onClick={() => alert('Continue with GitHub')}
      >
        Login with GitHub
      </button> */}
    </div>
  );
};

export default SocialLoginButtons;
