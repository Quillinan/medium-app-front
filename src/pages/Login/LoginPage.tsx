import LoginForm from '@components/forms/LoginForm/LoginForm';
import SocialLoginButtons from '@components/buttons/SocialLoginButtons/SocialLoginButtons';

const LoginPage: React.FC = () => {
  return (
    <div className='flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8'>
      <img
        className='mx-auto h-10 w-auto'
        src='https://tailwindcss.com/plus-assets/img/logos/mark.svg?color=indigo&shade=600'
        alt='Your Company'
      />

      <div className='sm:mx-auto sm:w-full sm:max-w-sm'>
        <h2 className='mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900'>
          Sign in to your account
        </h2>
      </div>

      <div className='mt-10 sm:mx-auto sm:w-full sm:max-w-sm'>
        <LoginForm />
        <div className='mt-6'>
          <SocialLoginButtons />
        </div>
      </div>
    </div>
  );
};

export default LoginPage;
