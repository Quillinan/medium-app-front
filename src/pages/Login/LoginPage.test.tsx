import { render, screen } from '@testing-library/react';
import LoginPage from './LoginPage';
import LoginForm from '@components/LoginForm/LoginForm';
import SocialLoginButtons from '@components/SocialLoginButtons/SocialLoginButtons';

vi.mock('@components/LoginForm/LoginForm', () => ({
  default: vi.fn(() => <div>Mocked LoginForm</div>),
}));

vi.mock('@components/SocialLoginButtons/SocialLoginButtons', () => ({
  default: vi.fn(() => <div>Mocked SocialLoginButtons</div>),
}));

describe('LoginPage', () => {
  it('should render the login page with logo, title, and form', () => {
    render(<LoginPage />);

    const logo = screen.getByAltText('Your Company');
    expect(logo).toBeInTheDocument();

    const title = screen.getByText(/Sign in to your account/i);
    expect(title).toBeInTheDocument();

    expect(LoginForm).toHaveBeenCalled();

    expect(SocialLoginButtons).toHaveBeenCalled();
  });
  it('should match snapshot', () => {
    const { asFragment } = render(<LoginPage />);

    expect(asFragment()).toMatchSnapshot();
  });
});
