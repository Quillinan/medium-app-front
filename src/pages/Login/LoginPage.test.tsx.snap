// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex min-h-full flex-1 flex-col justify-center px-6 py-12 lg:px-8"
  >
    <img
      alt="Your Company"
      class="mx-auto h-10 w-auto"
      src="https://tailwindui.com/plus/img/logos/mark.svg?color=indigo&shade=600"
    />
    <div
      class="sm:mx-auto sm:w-full sm:max-w-sm"
    >
      <h2
        class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
      >
        Sign in to your account
      </h2>
    </div>
    <div
      class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
    >
      <div>
        Mocked LoginForm
      </div>
      <div
        class="mt-6"
      >
        <div>
          Mocked SocialLoginButtons
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
